lstchain/datachecks/containers.py

```diff
@@ -173,35 +173,38 @@ def fill_event_wise_info(self, subrun_index, table, mask, geom,
                      bins=histogram_binnings.hist_dist0_intensity_gt_200)
         self.hist_dist0_intensity_gt_200 = counts
 
-        counts, _, _, _ = plt.hist2d(table['intensity'][mask],
-                                     table['width'][mask],
+        counts, _, _, _ = plt.hist2d(intensity, np.array(table['width'][mask]),
                                      bins=histogram_binnings.hist_width)
         self.hist_width = counts
 
-        counts, _, _, _ = plt.hist2d(table['intensity'][mask],
-                                     table['length'][mask],
+        counts, _, _, _ = plt.hist2d(intensity, np.array(table['length'][mask]),
                                      bins=histogram_binnings.hist_length)
         self.hist_length = counts
 
-        counts, _, _, _ = plt.hist2d(table['intensity'][mask],
-                                     table['skewness'][mask],
+        counts, _, _, _ = plt.hist2d(intensity,
+                                     np.array(table['skewness'][mask]),
                                      bins=histogram_binnings.hist_skewness)
         self.hist_skewness = counts
 
-        intercept = table['intercept'][mask]
         counts, _, _, _ = \
-            plt.hist2d(intensity, intercept,
+            plt.hist2d(intensity, np.array(table['intercept'][mask]),
```

Comment:
is tables a pandas series? if yes, `table.loc[mask, 'intercept'].to_numpy()` should be more efficient.